src/tdastro/astro_utils/snia_utils.py

```diff
@@ -0,0 +1,195 @@
+    """
+
+    cosmo = FlatLambdaCDM(H0=H0, Om0=Omega_m)
+    distmod = cosmo.distmod(redshift).value
+
+    return distmod
+
+
+class HostmassX1Func(FunctionNode):
+    """A wrapper class for the _hostmass_x1func() function.
+
+    Attributes
+    ----------
+    kind : `str`
+        The distance type for the Equivalency as defined by
+        astropy.cosmology.units.redshift_distance.
+
+    Parameters
+    ----------
+    hostmass : function or constant
+        The function or constant providing the hostmass value.
+    skewness : constant
+        Skewness parameter that defines the skewed normal distribution.
+    **kwargs : `dict`, optional
+        Any additional keyword arguments.
+    """
+
+    def __init__(self, hostmass, **kwargs):
+        # Call the super class's constructor with the needed information.
+        super().__init__(
+            func=_hostmass_x1func,
+            hostmass=hostmass,
+            **kwargs,
+        )
+
+class X0FromDistMod(FunctionNode):
+    """A wrapper class for the _x0_from_distmod() function.
+
+    Attributes
+    ----------
+    kind : `str`
+        The distance type for the Equivalency as defined by
+        astropy.cosmology.units.redshift_distance.
+
+    Parameters
+    ----------
+    distmod : function or constant
+        The function or constant providing the distance modulus value.
+    x1 : function or constant
+        The function or constant providing the x1 value.
+    c : function or constant
+        The function or constant providing the c value.
+    alpha : function or constant
+        The function or constant providing the alpha value.
+    beta : function or constant
+        The function or constant providing the beta value.
+    m_abs : function or constant
+        The function or constant providing the m_abs value.
+    **kwargs : `dict`, optional
+        Any additional keyword arguments.
+    """
+
+    def __init__(self, distmod, x1, c, alpha, beta, m_abs, **kwargs):
+        # Call the super class's constructor with the needed information.
+        super().__init__(
+            func = _x0_from_distmod,
+            distmod = distmod,
+            x1 = x1,
+            c = c,
+            alpha = alpha,
+            beta = beta,
+            m_abs = m_abs,
+            **kwargs,
+        )
+
+class DistModFromRedshift(FunctionNode):
+    """A wrapper class for the _distmod_from_redshift() function.
+
+    Attributes
+    ----------
+    kind : `str`
+        The distance type for the Equivalency as defined by
+        astropy.cosmology.units.redshift_distance.
+
+    Parameters
+    ----------
+    redshift : function or constant
+        The function or constant providing the redshift value.
+    H0 : constant
+        The Hubble constant.
+    Omega_m : constant
+        The matter density Omega_m.
+    **kwargs : `dict`, optional
+        Any additional keyword arguments.
+    """
+
+    def __init__(self, redshift,H0=73.,Omega_m=0.3, **kwargs):
```

Comment:
Do you expect to have runs where `H0` and `Omega_m` are generated from some distributions? Just curious.

Reply:
I think we would always specify the cosmological parameters from a simulation perspective. However if we want to do something like inferring the cosmological parameters using simulations to match the data, we would generate those from some distributions.